homeassistant/components/sensor/airvisual.py

```diff
@@ -95,60 +102,63 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     if city and state and country:
         _LOGGER.debug(
             "Using city, state, and country: %s, %s, %s", city, state, country)
+        location = ','.join((city, state, country))
         data = AirVisualData(
-            pav.Client(api_key), city=city, state=state, country=country,
+            Client(api_key), city=city, state=state, country=country,
             show_on_map=show_on_map)
     else:
         _LOGGER.debug(
             "Using latitude and longitude: %s, %s", latitude, longitude)
+        location = ','.join((str(latitude), str(longitude)))
         data = AirVisualData(
-            pav.Client(api_key), latitude=latitude, longitude=longitude,
+            Client(api_key), latitude=latitude, longitude=longitude,
             radius=radius, show_on_map=show_on_map)
 
     data.update()
+
     sensors = []
     for locale in monitored_locales:
         for sensor_class, name, icon in SENSOR_TYPES:
-            sensors.append(globals()[sensor_class](data, name, icon, locale))
+            sensors.append(classes[sensor_class](
+                data,
+                name,
+                icon,
+                locale,
+                adler32(location.encode('utf-8'))
```

Comment:
Why would you hash it?

Reply:
In this case, excluding some sort of hashing would result in unique IDs that look like this:

```yaml
sensor.allergy_index_yesterday:
  platform: pollen
  unique_id: 40.015275,-105.0582147_allergy_index_yesterday
```

...or...

```yaml
sensor.allergy_index_yesterday:
  platform: pollen
  unique_id: sao_paolo,sao_paolo,brazil_allergy_index_yesterday
```

I wasn't a fan of how ugly those names are, but if you think hashing is unnecessary, happy to go this route.

Reply:
I think that is fine.